File: backend/dataset/reverse/reverse_4.py
class Solution:
    def reverse(self, x: int) -> int:
        a = list(str(abs(x)))
        b = 0
        for i in range(len(a)):
            b += int(a[i]) * (10 ** i)
        
        if b < -(2**31) or b > ((2**31) - 1):
            return 0
        
        if x < 0:
            return (-1) * b
        else:
            return b

File: backend/CDataset/reverse/reverse_4.cpp
int reverse(int x) {
    int ans = 0, copy = abs(x);

    while (copy > 0) {
        if (INT32_MAX / 10 < ans) return 0;  // check overflow
        ans = ans * 10 + copy % 10;
        copy /= 10;
    }

    return x < 0 ? -ans : ans;
}
